rotkehlchen/chain/evm/decoding/weth/decoder.py

```diff
@@ -79,8 +94,20 @@ def _decode_deposit_event(self, context: DecoderContext) -> DecodingOutput:
                 event.event_subtype = HistoryEventSubType.DEPOSIT_ASSET
                 event.notes = f'Wrap {deposited_amount} {self.base_asset.symbol} in {self.wrapped_token.symbol}'  # noqa: E501
                 out_event = event
+            elif (
+                event.location == Location.SCROLL and
```

Comment:
same here .... you don't do special logic with if/else/if in base class when we are using inheritance. Come on .. this is programming 101